addons/survey/views/survey_templates_print.xml

```diff
@@ -16,14 +16,14 @@
                     <div role="form">
                         <fieldset disabled="disabled">
                             <t t-foreach='survey.question_and_page_ids' t-as='question'>
-                                <t t-if="question.is_page">
+                                <t t-if="question.is_page and any(q.id in questions_to_display for q in question.question_ids)">
                                     <hr t-if="question != survey.page_ids[0]" />
                                     <div class="o_page_header">
                                         <h1 t-field='question.title' />
                                         <div t-if="question.description" t-field='question.description' class="oe_no_empty"/>
                                     </div>
                                 </t>
-                                <t t-if="not question.is_page and not answer or (question in answer.predefined_question_ids)" >
+                                <t t-if="not question.is_page and not answer or (question.id in answer.predefined_question_ids.ids and question.id in questions_to_display)" >
```

Comment:
Why not pass the resultset to the template instead of the ids?
It would simplify the code here.

Would be:
```or (question in (answer.predefined_question_ids & questions_to_display))```